node/service/src/chain_spec/mandala.rs

```diff
@@ -45,25 +54,42 @@ pub fn parachain_dev_testnet_config(mnemonic: Option<&str>) -> Result<ChainSpec,
 	dev_testnet_config_from_chain_id("mandala-pc-dev", mnemonic)
 }
 
+fn generate_evm_address<W: Wordlist>(phrase: &str, index: u32) -> H160 {
+	let derivation_path =
+		coins_bip32::path::DerivationPath::from_str(&format!("{}{}", DEFAULT_DERIVATION_PATH_PREFIX, index))
+			.expect("should parse the default derivation path");
+	let mnemonic = Mnemonic::<W>::new_from_phrase(phrase).unwrap();
+
+	let derived_priv_key = mnemonic.derive_key(&derivation_path, None).unwrap();
+	let key: &SigningKey = derived_priv_key.as_ref();
+	let secret_key: SigningKey = SigningKey::from_bytes(&key.to_bytes()).unwrap();
+	let verify_key: VerifyingKey = secret_key.verifying_key();
+
+	let point: &K256PublicKey = &verify_key.to_encoded_point(false);
+	let public_key = point.to_bytes();
+
+	let hash = keccak256(&public_key[1..]);
+	H160::from_slice(&hash[12..])
+}
+
+fn keccak256<S>(bytes: S) -> [u8; 32]
+where
+	S: AsRef<[u8]>,
+{
+	let mut output = [0u8; 32];
+	let mut hasher = Keccak::v256();
+	hasher.update(bytes.as_ref());
+	hasher.finalize(&mut output);
+	output
+}
+
 fn get_evm_accounts(mnemonic: Option<&str>) -> Vec<H160> {
 	let phrase = mnemonic.unwrap_or("fox sight canyon orphan hotel grow hedgehog build bless august weather swarm");
-
 	let mut evm_accounts = Vec::new();
-	// Child key at derivation path: m/44'/60'/0'/0/{index}
 	for index in 0..10u32 {
-		let wallet = MnemonicBuilder::<English>::default()
-			.phrase(phrase)
-			.index(index)
-			.unwrap()
-			.build()
-			.unwrap();
-		evm_accounts.push(wallet.address());
-		log::info!(
-			"index: {:?}, private_key: {:x?} address: {:?}",
-			index,
-			hex::encode(wallet.signer().to_bytes()),
-			wallet.address()
-		);
+		let addr = generate_evm_address::<English>(phrase, index);
+		evm_accounts.push(addr);
+		log::info!("index: {:?}, evm address: {:?}", index, addr);
```

Comment:
I don't know why, the log can't be printed here

Reply:
e2e test can print: https://github.com/AcalaNetwork/Acala/runs/5827227801?check_suite_focus=true
<img width="581" alt="image" src="https://user-images.githubusercontent.com/1088525/161727266-8c3ece1d-f5b5-4ff8-b35b-6595805f4496.png">

I've checked that old ether-rs result is the same compare to our current implementation, so it wouldn't break anything.